### PR TITLE
WIP Import from cli

### PR DESCRIPTION
**Pull request by ildyria**
I found this lovely work by @Spotlight0xff  :smiley:

### CODE REVIEW

**Comment by ildyria** [comment]
I did not test this. @d7415 can you have a look ?

**Comment by kamil4** [comment]
Yeah, with the work that I'm planning, there should be no need for messing around with `owner_id`. Imported photos will simply inherit the ownership from the album.

File: app/Console/Commands/import.php [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+				'id',
+				'title',
+				'parent_id'
+			]);
+			$this->print_album($albums, 0 /* parent */, 0 /* indent */);
+			do {
+				if ($album_title != null) {
+					$this->line('No album '.$album_title.' found. Try again.');
+				}
+				$album_title = $this->ask('album to insert into (title)?');
```
Comment (d7415) [inline app/Console/Commands/import.php]: If we're providing a list of IDs with the titles, wouldn't it be easier to ask for the ID? Also potentially less typing, less chance of special characters...

File: app/Console/Commands/import.php [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+	/**
+	 * Execute the console command.
+	 *
+	 * @return mixed
+	 */
+	public function handle()
+	{
+		$directory = $this->argument('dir');
+		$flatten = $this->option('flatten');
```
Comment (d7415) [inline app/Console/Commands/import.php]: Good idea, but unused.

**Comment by ildyria** [comment]
I am also thinking if we could not re-use the following : https://github.com/LycheeOrg/Lychee-Laravel/blob/master/app/Http/Controllers/ImportController.php#L168 or so we don't have code duplication.

File: app/Console/Commands/import.php (hunk 1 above)
Comment (kamil4) [reply app/Console/Commands/import.php]: Also, I don't think that titles need to be unique...

**Comment by Spotlight0xff** [comment]
Hey guys!

Yeah this was written in order to allow me to sync my google photos to the lychee instance.
I wanted a way to just sync all the photos (which are downloaded using [timeliner](https://github.com/mholt/timeliner), with subfolders corresponding to the years/months) to a single album.
I don't have experience writing PHP-Laravel code, so the code is definitely not ideal.

Very cool that you found it and consider it to include it somehow though!

File: app/Console/Commands/import.php (hunk 1 above)
Comment (Spotlight0xff) [reply app/Console/Commands/import.php]: Definitely! It was more for testing, not meant to be production-ready :)

**Comment by alex-phillips** [comment]
FWIW, I'm working on a standalone CLI for lychee (both API versions supported) that allows uploading, tagging, searching, filtering, etc. of all photos in lychee. I will update the README with more info, but you can find the repo here: https://github.com/alex-phillips/lychee-cli

**Comment by ildyria** [comment]
Niiiice :open_mouth:

**Comment by ildyria** [comment]
This has been fixed with https://github.com/LycheeOrg/Lychee-Laravel/pull/427 I believe.

**Pull request closed without merging**